admin/create-theme/theme-locale.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/*
+* Locale related functionality
+*/
+class CBT_Theme_Locale {
+
+	/**
+	 * Escape a string for localization.
+	 *
+	 * @param string $string The string to escape.
+	 * @return string The escaped string.
+	 */
+	public static function escape_string( $string ) {
+		// Avoid escaping if the text is not a string.
+		if ( ! is_string( $string ) ) {
+			return $string;
+		}
+
+		// Check if the text is already escaped.
+		if ( str_starts_with( $string, '<?php echo' ) ) {
+			return $string;
+		}
+
+		$string = addcslashes( $string, "'" );
+		return "<?php echo __('" . $string . "', '" . wp_get_theme()->get( 'TextDomain' ) . "');?>";
+	}
+
+	/**
+	 * Get a replacement pattern for escaping the text from the html content of a block.
+	 *
+	 * @param string $block_name The block name.
+	 * @return string|array|null The regex pattern to match the content that needs to be escaped.
```

Comment:
Looks like it always returns an array (which is my 'druthers), so just `array|null` right?

Reply:
updated